src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -69,18 +69,26 @@ default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getName());
         if (!getPhone().isPrivate()) {
-            builder.append(" Phone: ").append(getPhone());
+            builder.append(getPrintableString(getPhone()));
         }
         if (!getEmail().isPrivate()) {
-            builder.append(" Email: ").append(getEmail());
+            builder.append(getPrintableString(getPhone()));
         }
         if (!getAddress().isPrivate()) {
-            builder.append(" Address: ").append(getAddress());
+            builder.append(getPrintableString(getPhone()));
         }
         builder.append(" Tags: ");
         for (Tag tag : getTags()) {
             builder.append(tag);
         }
         return builder.toString();
     }
+    
+    static String getPrintableString(Printable... printables){
+    	String printDetails = "";
+    	for (Printable p : printables){
```

Comment:
You can use the java StringBuilder shown above in your code, or look at StringJoiner and String.Join options. BTW you missed out the comma and spaces for the string.